pandas/tools/merge.py

```diff
@@ -1411,6 +1412,10 @@ def __init__(self, objs, axis=0, join='outer', join_axes=None,
             sample = objs[0]
         self.objs = objs
 
+        # Check for string axis parameter
+        if isinstance(axis, str):
```

Comment:
the check for string is not needed, this is already handled inside get_axis_number, so you can just do `axis = get_axis_number(axis)`